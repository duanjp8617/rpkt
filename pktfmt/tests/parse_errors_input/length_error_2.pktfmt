packet FakeProto {
    header = [
        field_1 = Field {
            bit = 8, 
            default = 4,
            gen = false,
        },
        field_2 = Field {
            bit = 8, 
            default = @5,
            gen = false
        },
        field_3 = Field {
            bit = 8, 
        }
    ],
    length = [
        header_len =field_1,
        payload_len =field_2*3,
    ]
}

