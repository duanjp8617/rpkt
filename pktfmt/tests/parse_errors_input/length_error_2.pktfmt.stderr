ast error
at /home/djp/pkt-rs/pktfmt/tests/parse_errors_input/length_error_2.pktfmt 17:5-20:5
         ~~~~~~~~~~
17 |     length = [
18 |         header_len =field_1,
19 |         payload_len =field_2*3,
20 |     ]
     ^^^^^
length error 2:
field field_2 used for computing the payload_len can not have a fixed default value
